tree/timer_test.go

```diff
@@ -0,0 +1,65 @@
+package tree
+
+import (
+	"testing"
+	"time"
+
+	"github.com/libp2p/go-libp2p-backoff/policy"
+)
+
+type TestBackoffPolicy struct{}
+
+func (TestBackoffPolicy) NewBackoffState() policy.BackoffState {
+	return &testBackoffState{}
+}
+
+type testBackoffState struct {
+	backingOff bool
+}
+
+func (s *testBackoffState) Clear(now time.Time) {
+	s.backingOff = false
+}
+
+func (s *testBackoffState) Backoff(now time.Time) {
+	s.backingOff = true
+}
+
+func (s *testBackoffState) TimeToClear(now time.Time) time.Duration {
+	if s.backingOff {
+		return time.Second
+	} else {
+		return time.Duration(0)
+	}
+}
+
+func TestTimerGC1(t *testing.T) {
+	// create root timer
+	root := NewBackoffTreeTimer(nil, TestBackoffPolicy{})
+	// create a child timer, which is cleared on init
+	root.Subtimer("child", TestBackoffPolicy{})
+	if root.NumChildren() != 1 {
+		t.Errorf("expecting one child")
+	}
+	// run GC on the root (should remove the child timer)
+	if root.GC() != 0 {
+		t.Errorf("GC did not remove child")
+	}
+}
+
+func TestTimerGC2(t *testing.T) {
```

Comment:
a test with two children subtimers active in different go-routines would be potentially useful to find races.